src/tree/gpu_hist/histogram.cu

```diff
@@ -145,40 +145,84 @@ __global__ void SharedMemHistKernel(EllpackDeviceAccessor matrix,
   }
 }
 
+template <typename GradientSumT>
+void InitGradientHistogram(int device_idx, int n_bins, HistogramLaunchConfig* config) {
```

Comment:
Prefer returning by value directly instead of via pointer. This is the functional programming "avoid side-effects" style.

Reply:
No longer applies.